### PR TITLE
Search feature

### PR DESCRIPTION
**Pull request by ghemDD**
The following changes come with this PR : 

-  Add search feature : find songs susceptible to feature in a playlist, find playlists of your choice
-  Implement ProfileViewModel to separate the concerns and be consistent with the actual architecture of the app
-  Correct some UI problems : playlist/song cardviews have fixed size now, more constrast between cardviews and background theme
- Correct some issues linked to Survival Mode : the number of lives is now accurately updated

Issues : #160 #158

### CODE REVIEW

**Review by JorisMonnet — APPROVED** [review]
Some Test are commented but otherwise good changes and some abreviation are not explicits

File: app/src/androidTest/java/ch/epfl/sdp/blindwar/ProfileFragmentTest.kt [hunk 1]
```diff
@@ -51,13 +53,11 @@ class ProfileFragmentTest : TestCase() {
 
     @Test
     fun testStatisticsButton() {
-
         launchFragmentInContainer<ProfileFragment>()
             closeSoftKeyboard()
-            onView(withId(R.id.statsButton))
+            onView(withId(R.id.statsBtn))
```
Comment (Marty99-dev) [inline app/src/androidTest/java/ch/epfl/sdp/blindwar/ProfileFragmentTest.kt:58]: We should not use abbreviation ! Must change this.

File: app/src/androidTest/java/ch/epfl/sdp/blindwar/ProfileFragmentTest.kt [hunk 2]
```diff
@@ -104,15 +140,15 @@ class ProfileFragmentTest : TestCase() {
         launchFragmentInContainer<ProfileFragment>()
         Thread.sleep(2000)
         val device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
-        onView(withId(R.id.statsButton))
+        onView(withId(R.id.statsBtn))
             .perform(click())
         device.pressBack()
-        onView(withId(R.id.historyButton))
+        onView(withId(R.id.historyBtn))
```
Comment (Marty99-dev) [inline app/src/androidTest/java/ch/epfl/sdp/blindwar/ProfileFragmentTest.kt:146]: Same.

File: app/src/main/java/ch/epfl/sdp/blindwar/data/music/MockMusicdataSource.kt [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package ch.epfl.sdp.blindwar.data.music
+
+import ch.epfl.sdp.blindwar.data.music.metadata.MusicMetadata
+import ch.epfl.sdp.blindwar.data.music.metadata.URIMusicMetadata
+import ch.epfl.sdp.blindwar.game.util.Tutorial
+
+object MockMusicdataSource {
```
Comment (Marty99-dev) [inline app/src/main/java/ch/epfl/sdp/blindwar/data/music/MockMusicdataSource.kt:7]: Nice Mock class. Usefull for test.

File: app/src/main/java/ch/epfl/sdp/blindwar/database/UserDatabase.kt [hunk 4]
```diff
@@ -136,6 +135,33 @@ object UserDatabase {
 //        ref.child(User.VarName.birthdate.name).setValue(user.birthdate)
 //    }
 
+    fun setFirstName(uid: String, fn: String) {
```
Comment (Marty99-dev) [inline app/src/main/java/ch/epfl/sdp/blindwar/database/UserDatabase.kt:138]: We should discuss if we should use inline function definition

**Review by Marty99-dev — APPROVED** [review]

**Pull request merged**